packages/rest/test/integration/rest.server.integration.ts

```diff
@@ -262,6 +269,119 @@ describe('RestServer (integration)', () => {
     expect(response.get('Access-Control-Allow-Credentials')).to.equal('true');
   });
 
+  it('exposes "GET /openapi.json" with openApiSpec.servers', async () => {
+    const server = await givenAServer({
+      rest: {
+        port: 0,
+        openApiSpec: {
+          servers: [{url: 'http://127.0.0.1:8080'}],
+        },
+      },
+    });
+    const greetSpec = {
+      responses: {
+        200: {
+          content: {'text/plain': {schema: {type: 'string'}}},
+          description: 'greeting of the day',
+        },
+      },
+    };
+    server.route(new Route('get', '/greet', greetSpec, function greet() {}));
+
+    const response = await createClientForHandler(server.requestHandler).get(
+      '/openapi.json',
+    );
+    expect(response.body).to.containDeep({
+      openapi: '3.0.0',
+      info: {
+        title: 'LoopBack Application',
+        version: '1.0.0',
+      },
+      servers: [{url: 'http://127.0.0.1:8080'}],
+      paths: {
+        '/greet': {
+          get: {
+            responses: {
+              '200': {
+                content: {
+                  'text/plain': {
+                    schema: {type: 'string'},
+                  },
+                },
+                description: 'greeting of the day',
+              },
+            },
+          },
+        },
+      },
+    });
+  });
+
+  it('exposes "GET /openapi.json" with openApiSpec.setServersFromRequest', async () => {
+    const server = await givenAServer({
+      rest: {
+        port: 0,
+        openApiSpec: {
+          setServersFromRequest: true,
+        },
+      },
+    });
+    const greetSpec = {
+      responses: {
+        200: {
+          content: {'text/plain': {schema: {type: 'string'}}},
+          description: 'greeting of the day',
+        },
+      },
+    };
+    server.route(new Route('get', '/greet', greetSpec, function greet() {}));
```

Comment:
I believe this route is not needed to verify that "setServersFromRequest" is handled correctly, could you please remove it?